src/storage-types/src/errors.rs

```diff
@@ -71,7 +71,21 @@ impl DecodeError {
 
 impl Display for DecodeError {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{} (original bytes: {:x?})", self.kind, self.raw)
+        // See if we can output the bytes that failed to decode as a string.
+        let str_repr = std::str::from_utf8(&self.raw).ok();
+        let bytes_repr = self
+            .raw
+            .iter()
+            .map(|b| format!("{:02x}", b))
+            .collect::<String>();
```

Comment:
This looks fairly allocation heavy—it’s an allocation per byte, right? `hexencode::encode` probably does something smarter.